### DISCUSSION TITLE
Refreshing JWT

### DISCUSSION THREAD

**Discussion by ycd** [fastapi/fastapi#7359] [Questions] [answered]
Is it possible to refresh tokens? In documentation, it provides an example of how to create a bearer token with a expire time but not how to refresh the token. 

In flask it has a package for it: [Flask-JWT-Extended](https://flask-jwt-extended.readthedocs.io/en/stable/refresh_tokens/?highlight=refresh)

I looked up source code of  FastAPI and [Python-JOSE](https://github.com/mpdavis/python-jose) couldn't find anything.

**Comment by shidenko97** [comment, marked as answer]
As I can see - it's impossible now. But you can recreate token by the same data and it will work like update. 

For example:
```
import datetime
from time import sleep

from jose import jwt
from jose.exceptions import ExpiredSignatureError


if __name__ == "__main__":

    data = {"bearer": "It's an example of bearer"}
    secret = "Secret key"
    expired = 5

    data["exp"] = datetime.datetime.utcnow() + datetime.timedelta(
        seconds=expired
    )

    token = jwt.encode(data, secret)

    print(token)

    sleep(expired + 1)

    try:
        decrypted = jwt.decode(token, secret)
    except ExpiredSignatureError:
        data["exp"] = datetime.datetime.utcnow() + datetime.timedelta(
            seconds=expired
        )
        token = jwt.encode(data, secret)

        print(token)

        decrypted = jwt.decode(token, secret)

    print(decrypted)
```

**Comment by tiangolo** [comment]
Thanks for the help @shidenko97 and thanks for closing the issue @ycd! 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
